client/settings/fraud-protection/advanced-settings/rule-toggle.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * External dependencies
+ */
+import React, { useContext, useEffect, useState } from 'react';
+import { __ } from '@wordpress/i18n';
+import { CheckboxControl, Icon, ToggleControl } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import './../style.scss';
+import Tooltip from '../../../components/tooltip';
+import FraudPreventionSettingsContext from './context';
+
+const FraudProtectionRuleToggle = ( {
+	setting,
+	label,
+	helpText,
+	children,
+} ) => {
+	const {
+		advancedFraudProtectionSettings,
+		setAdvancedFraudProtectionSettings,
+	} = useContext( FraudPreventionSettingsContext );
+
+	const [ toggleState, setToggleState ] = useState( false );
+	const [ checkState, setCheckState ] = useState( false );
+
+	// Set initial states from saved settings.
+	useEffect( () => {
+		if (
+			advancedFraudProtectionSettings &&
+			advancedFraudProtectionSettings[ setting ]
+		) {
+			setToggleState(
+				advancedFraudProtectionSettings[ setting ].enabled
+			);
+			setCheckState( advancedFraudProtectionSettings[ setting ].block );
+		}
+	}, [
+		advancedFraudProtectionSettings,
+		setToggleState,
+		setCheckState,
+		setting,
+	] );
+
+	// Set global object values from input changes.
+	useEffect( () => {
+		if (
+			advancedFraudProtectionSettings &&
+			advancedFraudProtectionSettings[ setting ]
+		) {
+			advancedFraudProtectionSettings[ setting ].enabled = toggleState;
+			advancedFraudProtectionSettings[ setting ].block = checkState;
+			setAdvancedFraudProtectionSettings(
```

Comment:
In order to prevent immutability issues, I'd suggest changing the update approach to something like this:

```js
setAdvancedFraudProtectionSettings( ( settings ) => ( {
  ...settings,
  [ setting ]: {
    ...settings[ setting ],
    block: checkState,
    enabled: toggleState,
  },
} ) );
```

Reply:
It broke everything after I switched to that approach, so I've reverted it back. It'll also be safe to use as is.